### DISCUSSION TITLE
ModuleNotFoundError when dockerizing obfuscated scripts

### DISCUSSION THREAD

**Discussion by Waschbrettwade** [dashingsoft/pyarmor#1776] [Q&A]
Hi there,

I am obfuscating dummy Python scripts and they run fine. However, when building a Docker image, I get

```
Traceback (most recent call last):
  File "/app/wrapper.py", line 2, in <module>
    from pyarmor_runtime_000000 import __pyarmor__
  File "/app/pyarmor_runtime_000000/__init__.py", line 2, in <module>
    from .pyarmor_runtime import __pyarmor__
ModuleNotFoundError: No module named 'pyarmor_runtime_000000.pyarmor_runtime'
```

The wrapper.py looks like this:

```
from pyarmor_runtime_000000 import __pyarmor__
__pyarmor__(__name__, __file__, b'PY000000\x00\x03\x0b\x00\xa7\r\r......
```

I went into the container, the module is definitely there. Why can't it be imported in the Docker image, but outside? How to fix this? Thank you very much!

**Comment by jondy** [comment]
Run the obfuscated scripts in docker container need `--platform alpine.xxx`, it doesn't work if using native linux runtime package.

**Reply by Waschbrettwade** [reply]
Thank you for you quick answer!

I have tried obfuscating scripts on Ubuntu, but the scripts didn't work at all, no matter which platform I chose. 

Trying on Windows, I can successfully turn the scripts into obfuscated one, using --platform windows.x86_64. Building a Docker image using that Dockerfile

```
FROM python:3.9-bookworm

WORKDIR /app

COPY . /app

ENV PYTHONPATH="/app:${PYTHONPATH}"

RUN pip install --trusted-host pypi.python.org --upgrade pip

RUN pip install --trusted-host pypi.python.org -r requirements.txt

EXPOSE 8501

CMD ["python", "wrapper.py"]
```

works. But running it also gets me the exact same error message as in my post above.

**Comment by jondy** [comment]
Check this
https://pyarmor.readthedocs.io/en/latest/topic/obfuscated-script.html

**Comment by Waschbrettwade** [comment]
I could not resolve this issue, but, at least for me, this is not relevant anymore. Thank you for your effort, @jondy
